core/templates/dev/head/pages/exploration_editor/editor_tab/state_editor_solution.html

```diff
@@ -0,0 +1,61 @@
+<div ng-controller="StateSolution">
+  <div ng-if="!isCurrentInteractionLinear() &&
```

Comment:
Is "isCurrentInteractionLinear" already included in "canSupportSolution"? If so, maybe remove that check, and also add a backend test which ensures that all interactions that can support solutions are not linear, in order to ensure that this invariant is preserved in the future.